### PR TITLE
Update doc for service principal certificate authentication

### PR DESCRIPTION
**Pull request by jiasli**
Update doc according to https://github.com/Azure/azure-cli/issues/30283

### CODE REVIEW

File: docs-ref-conceptual/authenticate-azure-cli-service-principal.md [hunk 1]
```diff
@@ -18,32 +18,40 @@ about service principals, see [Work with Azure service principals using the Azur
...
+
+```azurecli-interactive
+az login --service-principal --username APP_ID --password CLIENT_SECRET --tenant TENANT_ID
+```
+
+Log in with certificate:
+
 ```azurecli-interactive
-az login --service-principal -u <app-id> -p <password-or-cert> --tenant <tenant>
+az login --service-principal --username APP_ID --certificate /path/to/cert.pem --tenant TENANT_ID
```
Comment (jiasli) [inline docs-ref-conceptual/authenticate-azure-cli-service-principal.md:39]: Full argument names should be preferred.

File: docs-ref-conceptual/azure-cli-sp-tutorial-3.md [hunk 2]
```diff
@@ -142,10 +140,7 @@ The output includes credentials that you must protect. Be sure that you do not i
 To sign in with a certificate, the certificate must be available locally as a PEM or DER file in ASCII format. PKCS#12 files (.p12/.pfx) don't work. When you use a PEM file, the **PRIVATE KEY** and **CERTIFICATE** must be appended together within the file. You don't need to prefix the path with an `@` like you do with other az commands.
 
 ```azurecli-interactive
-az login --service-principal \
-         --username myServicePrincipalID \
-         --tenant myOwnerOrganizationId \
-         --password /path/to/cert
+az login --service-principal --username APP_ID --certificate /path/to/cert.pem --tenant TENANT_ID
```
Comment (jiasli) [inline docs-ref-conceptual/azure-cli-sp-tutorial-3.md:143]: This is not a very long command, so no need to break it into several lines.

File: docs-ref-conceptual/azure-cli-sp-tutorial-3.md [hunk 3]
```diff
@@ -102,8 +102,6 @@ az keyvault secret download --file /path/to/cert.pfx \
                             --name CertName \
                             --encoding base64
 openssl pkcs12 -in cert.pfx -passin pass: -passout pass: -out cert.pem -nodes
-
-az login --service-principal -u "<myAppClientID>" -p cert.pem --tenant "<myTenantID>"
```
Comment (jiasli) [inline docs-ref-conceptual/azure-cli-sp-tutorial-3.md:106]: No need to mention `az login` command here. It is later mentioned in "Sign in with a service principal using a certificate" section.

File: docs-ref-conceptual/upcoming-breaking-changes.md [hunk 4]
```diff
@@ -63,6 +63,12 @@ The [db-up](https://github.com/Azure/azure-cli-extensions/tree/main/src/db-up) e
 
 - This command group has been deprecated and will be removed in next breaking change release(2.67.0). Use 'az network public-ip ddos-protection' instead.
 
+## profile
+
+### `az login`
+
+- Passing the service principal certificate with `--password` is deprecated and will be removed in version 2.67.0. Please use `--certificate` instead.
```
Comment (dcaro) [inline docs-ref-conceptual/upcoming-breaking-changes.md:70]:
```suggestion
- The `--password` parameter will not accept service principal certificate starting with version 2.67.0, use the `--certificate` instead.
```

File: docs-ref-conceptual/authenticate-azure-cli-service-principal.md [hunk 5]
```diff
@@ -18,32 +18,40 @@ about service principals, see [Work with Azure service principals using the Azur
...
-> read -sp "Azure password: " AZ_PASS && echo && az login --service-principal -u <app-id> -p $AZ_PASS --tenant <tenant>
+> read -sp "Azure password: " AZ_PASS && echo && az login --service-principal --username <app-id> --password $AZ_PASS --tenant <tenant>
 > ```
 >
 > Under PowerShell, use the `Get-Credential` cmdlet.
 >
 > ```powershell
 > $AzCred = Get-Credential -UserName <app-id>
-> az login --service-principal -u $AzCred.UserName -p $AzCred.GetNetworkCredential().Password --tenant <tenant>
+> az login --service-principal --username $AzCred.UserName --password $AzCred.GetNetworkCredential().Password --tenant <tenant>
```
Comment (dcaro) [inline docs-ref-conceptual/authenticate-azure-cli-service-principal.md:54]: @mikefrobbins any thoughts on using `$Cred.GetNetworkCredential` vs `$Cred.Password` ?

**Review by dcaro — APPROVED** [review]
Please look at my comments

File: docs-ref-conceptual/upcoming-breaking-changes.md (hunk 4 above)
Comment (jiasli) [reply docs-ref-conceptual/upcoming-breaking-changes.md:70]: Is the `the` before `--certificate` needed?

Comment (jiasli) [reply docs-ref-conceptual/upcoming-breaking-changes.md:70]: Maybe we can align with https://github.com/Azure/azure-cli/issues/30283?

**Review by evelyn-ys — APPROVED** [review]

**Review by bebound — APPROVED** [review]

**Review by jsntcy — APPROVED** [review]

**Pull request merged**